src/main/java/frc/bionic/Conversion.java

```diff
@@ -0,0 +1,200 @@
+   */
+  static public double wheelRevsToMetersTraveled(double revs, double wheelDiameterMeters)
+  {
+    double          circumferenceMeters = Math.PI * wheelDiameterMeters;
+    return circumferenceMeters * revs;
+  }
+
+  /**
+   * Convert wheel meters to revolutions traveled
+   *
+   * @param meters
+   *   Number of meters traveled
+   *
+   * @param wheelDiameterMeters
+   *   The diameter of the wheel
+   *
+   * @return
+   *   Equivalent number of revolutions traveled
+   */
+  static public double wheelMetersTraveledToRevs(double meters, double wheelDiameterMeters)
+  {
+    double          circumferenceMeters = Math.PI * wheelDiameterMeters;
+    return meters / circumferenceMeters;
+  }
+
+  /**
+   * Convert revolutions per minute to meters per second
+   *
+   * @param rpm
+   *   Revolutions per minute to convert
+   *
+   * @param wheelDiameterMeters
+   *   The diameter of the wheel
+   *
+   * @return
+   *   Revolutions per minute converted into meters per second
+   */
+  static public double rpmToMps(double rpm, double wheelDiameterMeters)
+  {
+    double          circumferenceMeters = Math.PI * wheelDiameterMeters;
+    return rpm * circumferenceMeters / 60;
+  }
+
+  /**
+   * Convert meters per second to revolutions per minute
+   *
+   * @param meters
+   *   Meters per second to convert
+   *
+   * @param wheelDiameterMeters
+   *   The diameter of the wheel
+   *
+   * @return
+   *   Meters per second converted into revolutions per minute
+   */
+  static public double mpsToRpm(double mps, double wheelDiameterMeters)
+  {
+    double          circumferenceMeters = Math.PI * wheelDiameterMeters;
+    return mps * 60 / circumferenceMeters;
+  }
+
+  /**
+   * Convert rotary encoder ticks to corresponding number of degrees of movement
+   *
+   * @param ticks
+   *   Number of ticks to convert
+   *
+   * @param ticksPerRev
+   *   Number of ticks per one full revolution of the encoder
+   *
+   * @return
+   *   Degrees of motion resulting from specified number of ticks
+   */
+  static public double rotaryEncoderTicksToDegrees(double ticks, double ticksPerRev)
+  {
+    return ticks * (360.0 / ticksPerRev);
+  }
+
+  /**
+   * Convert degrees of encoder movement to corresponding number of ticks
+   *
+   * @param degrees
+   *   Number of degrees to convert
+   *
+   * @param ticksPerRev
+   *   Number of ticks per one full revolution of the encoder
+   *
+   * @return
+   *   Ticks resulting from specified degrees of motion
+   */
+  static public double degreesToRotaryEncoderTicks(double degrees, double ticksPerRev)
+  {
+    return degrees / (360.0 / ticksPerRev);
+  }
+
+  /**
+   * Convert degrees to radians, first scaling degrees to to the range
+   * [ 0.0, 360.0 ].
+   *
+   * @param degrees
+   *   The number of degrees to convert to radians
+   *
+   * @return
+   *   The degrees converted to radians
+   */
+  static public double degreesToRadians(double degrees)
+  {
+    while (degrees > 360.0)
```

Comment:
Is this what wpilib does? Feels like there is a way to express these ranges without loops

Reply:
I don't know what they do. What I'd really like is more like the `mod` operator that operates on doubles. I'll look into it.

Reply:
It turns out that `mod` (`%`) does work on doubles in java, but encounters problems with floating point precision. I found a nice simple function that normalizes, and should work properly with all values.